avaframe/com4FlowPy/flowClass.py

```diff
@@ -389,16 +389,19 @@ def calc_distribution(self):
 
         if mass_to_distribute > 0 and count > 0:
             # local flux redistribution to eligible child cells
-            self.dist[self.dist > threshold] += mass_to_distribute / count
+            self.dist[self.dist >= threshold] += mass_to_distribute / count
             self.dist[self.dist < threshold] = 0
         if np.sum(self.dist) < self.flux and count > 0:
             # correction/flux conservation for potential rounding losses
-            self.dist[self.dist > threshold] += (self.flux - np.sum(self.dist)) / count
+            self.dist[self.dist >= threshold] += (self.flux - np.sum(self.dist)) / count
+        if np.sum(self.dist) > self.flux and count > 0:
+            # correction/flux conservation for potential rounding gains
+            self.dist[self.dist >= threshold] += (self.flux - np.sum(self.dist)) / count
```

Comment:
thx, correction for rounding gains also was missing, maybe you can simplify by handling both cases (rounding gains, rounding losses) in a single statement
```suggestion
        if np.sum(self.dist) != self.flux and count > 0:
            # correction/flux conservation for potential rounding losses or gains
            # (self.flux - np.sum(self.dist)) will either be negative or positive 
            # depending on the direction of the rounding error
            self.dist[self.dist >= threshold] += (self.flux - np.sum(self.dist)) / count
```

Reply:
Hi @theweathermanda : is the suggestion ok for you? Then you could commit it and I'll merge this PR...

Reply:
@fso42 Added a commit below for this change. Thanks!